cypress/support/commands.js

```diff
@@ -231,28 +235,45 @@ Cypress.Commands.add('empiricaLoginMultiPlayers', (playerKeys) => {
+  let url = `/?${urlParams.join('&')}`
+  if (enableVideoCall) {
+    url = url + `&videoCall=true`
+  } 
+  cy.visit(url, { log: false });
   cy.wait(300, { log: false })
   log.snapshot("before");
 
 
   //consent
-  cy.get("[test-player-id='player1']").contains("consent", { timeout: 5000, log: false });
-  cy.get("[test-player-id='player1']").contains("you may engage in video, audio, or text chat", { log: false });  // check IRB language present
-  cy.get("[test-player-id='player1']").contains("We may share recordings under a confidentiality agreement", { log: false });  // check IRB language present
-  cy.get("[test-player-id='player1']").contains("deliberation-study@wharton.upenn.edu", { log: false });  // check contact info present
-  cy.get("[test-player-id='player1']").find("button", { log: false }).contains("I AGREE", { log: false }).click({ log: false });
-  cy.scrollTo('bottom');
-  cy.get("[test-player-id='player2']").find("button", { log: false }).contains("I AGREE", { log: false }).click({ log: false });
-
-  // Login
-  cy.get("[test-player-id='player1']").contains("Enter your", { timeout: 5000, log: false, matchCase: false });
-  cy.get("[test-player-id='player1']").find("input[id='playerID']", { log: false }).click({ log: false }).type(playerKeys[0], { log: false });
-  cy.get("[test-player-id='player1']").find("button", { log: false }).contains("Enter", { log: false }).click({ log: false });
-  cy.get("[test-player-id='player2']").find("input[id='playerID']", { log: false }).click({ log: false }).type(playerKeys[1], { log: false });
-  cy.get("[test-player-id='player2']").find("button", { log: false }).contains("Enter", { log: false }).click({ log: false });
+  playerKeys.forEach(playerKey => {
+    cy.get(`[test-player-id='${playerKey}']`, { log: false }).then( $player =>{
+      // Consent
+      cy.wrap($player, { log: false }).contains("consent", { timeout: 5000, log: false });
+      
+      // Check IRB language present
+      cy.wrap($player, { log: false }).contains("you may engage in video, audio, or text chat", { log: false });  
+      cy.wrap($player, { log: false }).contains("We may share recordings under a confidentiality agreement", { log: false }); 
+      
+      // Check contact info present
+      cy.wrap($player, { log: false }).contains("deliberation-study@wharton.upenn.edu", { log: false });
+      
+      // Submit
+      cy.wrap($player, { log: false }).find("button", { log: false }).contains("I AGREE", { log: false }).click({ log: false });
+
+      // Login
+      cy.wrap($player, { log: false }).contains("Enter your", { timeout: 5000, log: false, matchCase: false });
+      cy.wrap($player, { log: false }).find("input", { log: false }).click({ log: false }).type(playerKey, { log: false });
+      cy.wrap($player, { log: false }).find("button", { log: false }).contains("Enter", { log: false }).click({ log: false });
+    } )
+  })
```

Comment:
I like this wrap syntax! It's pretty. Do you have to wrap it everytime or is it possible to define a variable and just run .contains on it?

Reply:
Good question! I don't know - I'll look it up.